### ISSUE TITLE
Neos\Neos\Domain\Model\Domain same schema different port

### ISSUE THREAD

**Issue by markuspfeifenberger** [neos/neos-development-collection#1622]
### Description

In Neos\Neos\Domain\Model\Domain it is not possible to have the same schema with different ports, cause the scheme is unique, e.g. this does not work, but is a reallive scenario

localhost:3000
localhost:3001
localhost:3002
...

Should the restriction not be a unique key on schema and port?

### Affected Versions
Neos 3.1.4

**Comment by aertmann** [comment]
Good point, the restriction is due to [UniqueEntityValidator](https://github.com/neos/flow-development-collection/blob/master/Neos.Flow/Classes/Validation/Validator/UniqueEntityValidator.php) being used in the SitesController [here](https://github.com/neos/neos-development-collection/blob/master/Neos.Neos/Classes/Controller/Module/Administration/SitesController.php#L415) and [here](https://github.com/neos/neos-development-collection/blob/master/Neos.Neos/Classes/Controller/Module/Administration/SitesController.php#L384)

To fix this the `@Flow\Identity` annotation needs to be added to the `scheme` and `port` properties in addition to the `hostname` [property](https://github.com/neos/neos-development-collection/blob/master/Neos.Neos/Classes/Domain/Model/Domain.php#L31). Also the `getCacheEntryIdentifier` needs to be adjusted to combine those three properties as the cache identifier.

Along with that a MySQL and a Postgres migration needs to be provided.

Depending if this is considered a bug or a feature, it either needs to go into 2.3 or master.
